src/PKSim.Core/Snapshots/Mappers/ObservedDataMapper.cs

```diff
@@ -45,9 +47,21 @@ private Task<SnapshotExtendedProperties> mapExtendedProperties(ModelExtendedProp
          return _extendedPropertiesMapper.MapToSnapshot(extendedProperties);
       }
 
-      public override Task<ModelDataRepository> MapToModel(SnapshotDataRepository snapshot)
+      public override async Task<ModelDataRepository> MapToModel(SnapshotDataRepository snapshot)
       {
-         throw new System.NotImplementedException();
+         var dataRepository = new ModelDataRepository
+         {
+            Name = snapshot.Name,
+            Description = snapshot.Description
+         };
+
+         dataRepository.Add(await _dataColumnMapper.MapToModel(snapshot.BaseGrid, dataRepository));
+
+         snapshot.Columns.Each(async x => dataRepository.Add(await _dataColumnMapper.MapToModel(x, dataRepository)));
```

Comment:
this is going to cause all mapping to wait for each other. In general, never do an await in a loop. Instead use Tasks.WhenAll 
(in our case, there is no io etc.. so it will effectively be the same but why not  use the expected pattern)